solutions/USACO feb17-circlecross.mdx

```diff
@@ -2,223 +2,80 @@
 id: usaco-712
 source: USACO Bronze 2017 February
 title: Why Did the Cow Cross the Road II
-author: Benjamin Qi
+author: Benjamin Qi, Danh Ta Chi Thanh
 ---
 
+## Solution 1
+
 [Official Analysis](http://www.usaco.org/current/data/sol_circlecross_bronze_feb17.html)
 
+## Solution 2
+
+Let the first occurrence of a letter be the cow's entry and the last occurrence her exit. Then, we count the number of pairs $(cow_1, cow_2)$ such that $cow_1$ enters before $cow_2$ and leaves somewhere between $cow_2$'s entry and exit. 
+
+In other words, we define $start_x$, $end_x$, as the entry and exit points of a $cow_x$, where $start_x$ is the index of the first occurrence of $cow_x$ and $end_x$ is the index of the last occurrence of $cow_x$. Thus, we count the number of pairs of cows $(x, y)$ such that $start_x$ < $start_y$ < $end_x$ < $end_y$.
+
+**Time Complexity:** $\mathcal{O}(N^4)$ (where $N$ is the number of entries and exits)
+
+Alternatively, we can store the positions of entry points and exit points in another arrays in order to reduce the **Time Complexity** to  $\mathcal{O}(N^2)$. 
```

Comment:
```suggestion
Alternatively, we can store the positions of entry points and exit points in another arrays in order to reduce the **Time Complexity** to  $\mathcal{O}(N^2)$. 
```
```suggestion
Alternatively, we can store the positions of entry and exit points for each of the cows. Then, we can loop through all pairs of cows and see if they cross based on their entry and exit points. This runs in $\mathcal O(N^2)$.
```

Reply:
Thanks a lot, the explanation seems concise and accurate now.